src/spec-common/injectHeadless.ts

```diff
@@ -433,21 +442,45 @@ async function runPostCommand({ postCreate }: ResolverParameters, containerPrope
 			onDidChangeDimensions: postCreate.output.onDidChangeDimensions,
 		}, LogLevel.Info);
 		try {
-			infoOutput.event({
-				type: 'progress',
-				name: progressName,
-				status: 'running',
-				stepDetail: progressDetail
-			});
 			const remoteCwd = containerProperties.remoteWorkspaceFolder || containerProperties.homeFolder;
+			async function runSingleCommand(postCommand: string | string[], name?: string) {
+				const progressDetail = typeof postCommand === 'string' ? postCommand : postCommand.join(' ');
+				infoOutput.event({
```

Comment:
This is written to the progress file consumed by the Codespaces agent. Will that and the UI know how to deal with multiple entries? /cc @samruddhikhandale

Reply:
This is the UI, which Christof mentioned.

![image](https://user-images.githubusercontent.com/24955913/193662952-3a65c992-dde6-4807-aaa6-c4788a46c00e.png)

The agent code needs to be modified to handle postCommands running in parallel else it will show incorrect details.
Currently, the tooling is setup for handling sequential events 😓 

// cc @jkeech

Reply:
@jkeech and I talked it over and decided we could go ahead and land this as-is, and then do the follow up repair work depending on what the Codespaces experience is in a separate PR.